static/js/src/advantage/users/api.ts

```diff
@@ -0,0 +1,128 @@
+import { UserRole, Users } from "./types";
+
+type AccountUsersReponse = {
+  account_id: string;
+  name: string;
+  users: {
+    id: string;
+    name: string;
+    email: string;
+    user_role_on_account: "admin" | "technical" | "billing";
+    last_login_at: string;
+  }[];
+};
+
+type ParsedAccountUsersResponse = {
+  accountId: string;
+  organisationName: string;
+  users: Users;
+};
+
+const parseAccountsResponse = (
+  response: AccountUsersReponse
+): ParsedAccountUsersResponse => ({
+  accountId: response.account_id,
+  organisationName: response.name,
+  users: response.users.map((user) => ({
+    id: user.email,
+    name: user.name,
+    email: user.email,
+    role: user.user_role_on_account,
+    lastLoginAt: user.last_login_at,
+  })),
+});
+
+const requestAccountUsers = async (): Promise<ParsedAccountUsersResponse> => {
+  const response = await fetch(
+    `/advantage/account-users${window.location.search}`,
+    {
+      cache: "no-store",
+    }
+  );
+
+  return response.json().then(parseAccountsResponse);
+};
+
+const accountUserRequestInit: RequestInit = {
+  cache: "no-store",
+  credentials: "include",
+  headers: {
+    Accept: "application/json",
+    "Content-Type": "application/json",
+  },
+};
+
+const getAccountUserRequestUrl = (accountId: string, urlParams: string) =>
+  `/advantage/accounts/${accountId}/user${urlParams}`;
+
+type JSONResponse = {
+  data?: Record<string, unknown>;
+  error?: string;
+  errors?: string;
```

Comment:
2 error fields?

Reply:
yup, seems like the API isn't very consistent 😬